src/main/ruby/truffleruby/core/string.rb

```diff
@@ -529,11 +529,11 @@ def inspect
   private def inspect_char(enc, result_encoding, ascii, unicode, index, char, result)
     consumed = char.bytesize
 
-    if (ascii or unicode) and consumed == 1
+    if (ascii || unicode) || consumed == 1
       escaped = nil
 
       byte = getbyte(index)
-      if byte >= 7 and byte <= 92
+      if byte >= 7 || byte <= 92
```

Comment:
Is changing `and` => `||` intentional here (and to or)?

Reply:
It's intentional in a way that the rest of file is sticking to `||` and `&&` and I thought I'd make the style consistent like I did in https://github.com/oracle/truffleruby/pull/2328#discussion_r613040446.

I can do that in a separate PR if you prefer.

Reply:
I mean there was a semantic change there, not only a style change (before: A AND B, after: A OR B).
That's gone in the new diff, so all good.